source/common/ssl/context_impl.cc

```diff
@@ -797,5 +801,59 @@ int ServerContextImpl::sessionTicketProcess(SSL*, uint8_t* key_name, uint8_t* iv
   }
 }
 
+enum ssl_select_cert_result_t
+ServerContextImpl::selectTlsContext(const SSL_CLIENT_HELLO* ssl_client_hello) {
+  // This is currently a nop, since we only have a single cert, but this is where we will implement
+  // the certificate selection logic in #1319.
+  RELEASE_ASSERT(SSL_set_SSL_CTX(ssl_client_hello->ssl, tls_contexts_[0].ssl_ctx_.get()) != nullptr,
+                 "");
+  return ssl_select_cert_success;
+}
+
+void ServerContextImpl::TlsContext::addClientValidationContext(
```

Comment:
Nit: adding `ServerContextImpl::setSessionIdContext(TlsContext& ctx, ...)` and `ServerContextImpl::TlsContext::addClientValidationContext(...)` feels a bit inconsistent.